akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -35,3 +43,29 @@ trait SinkQueue[T] {
    */
   def pull(): Future[Option[T]]
 }
+
+sealed abstract class StreamCallbackStatus[+T]
+
+/**
+ * Contains types that is used as return types for async callbacks to streams
+ */
+object StreamCallbackStatus {
```

Comment:
I actually tried to make it compliant with SinkQueue#pull. Where we can do get result as T if steam is has something